configuration/uart.md

```diff
@@ -1,51 +1,111 @@
+
+| Primary UART | Default state of enable_uart flag |
+|--------------|-----------------------------------|
+| mini UART    | 0 |
+| first PL011 (UART0)       | 1 |
+
+## Disable Linux serial console
+
+By default, the primary UART is assigned to the Linux console. If you wish to use the primary UART for other purposes, you must reconfigure Raspbian. This can be done by using [raspi-config](raspi-config.md):
+
+1. Start raspi-config: `sudo raspi-config`.
+1. Select option 5 - interfacing options.
+1. Select option P6 - serial.
+1. At the prompt `Would you like a login shell to be accessible over serial?` answer 'No'
+1. At the prompt `Would you like the serial port hardware to be enabled?` answer 'Yes'
+1. Exit raspi-config and reboot the Pi for changes to take effect.
 
 ## UARTs and Device Tree
 
-Various UART Device Tree Overlay definitions can be found in the kernel github tree. The two most useful overlays are [`disable-bt`](https://github.com/raspberrypi/linux/blob/rpi-4.11.y/arch/arm/boot/dts/overlays/disable-bt-overlay.dts) and [`miniuart-bt`](https://github.com/raspberrypi/linux/blob/rpi-4.11.y/arch/arm/boot/dts/overlays/miniuart-bt-overlay.dts).
+Various UART Device Tree overlay definitions can be found in the [kernel GitHub tree](https://github.com/raspberrypi/linux). The two most useful overlays are [`disable-bt`](https://github.com/raspberrypi/linux/blob/rpi-4.19.y/arch/arm/boot/dts/overlays/disable-bt-overlay.dts) and [`miniuart-bt`](https://github.com/raspberrypi/linux/blob/rpi-4.19.y/arch/arm/boot/dts/overlays/miniuart-bt-overlay.dts).
 
-`disable-bt` disables the Bluetooth device and restores UART0/ttyAMA0 to GPIOs 14 and 15. It is also necessary to disable the system service that initialises the modem so it doesn't use the UART: `sudo systemctl disable hciuart`.
+`disable-bt` disables the Bluetooth device and makes the first PL011 (UART0) the primary UART. It is also necessary to disable the system service that initialises the modem, so it doesn't connect to the UART using `sudo systemctl disable hciuart`.
 
-`miniuart-bt` switches the Raspberry Pi 3 and Raspberry Pi Zero W Bluetooth function to use the mini UART (ttyS0), and restores UART0/ttyAMA0 to GPIOs 14 and 15. Note that this may reduce the maximum usable baudrate (see mini UART limitations below). It is also necessary to edit /lib/systemd/system/hciuart.service and replace ttyAMA0 with ttyS0, unless you have a system with udev rules that create /dev/serial0 and /dev/serial1. In this case, use /dev/serial1 instead because it will always be correct. If cmdline.txt uses the alias serial0 to refer to the user-accessible port, the firmware will replace it with the appropriate port whether or not this overlay is used.
+`miniuart-bt` switches the Bluetooth function to use the mini UART, and makes the first PL011 (UART0) the primary UART. Note that this may reduce the maximum usable baud rate (see mini UART limitations below).
```

Comment:
Does this overlay also lock the core clock frequency @pelwell ?

Reply:
No it doesn't - a fact mentioned in the README IIRC, but worth reiterating.

Reply:
I've added a note that the core clock needs set to a fixed frequency to this paragraph.